vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -550,11 +551,28 @@ void AdsImpl::SetConfirmationsIsReady(
   is_confirmations_ready_ = is_ready;
 }
 
-std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory() {
+std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory(
+    const AdsHistoryFilterType AdsHistoryFilterType) {
   std::map<uint64_t, std::vector<AdsHistory>> ads_history;
   base::Time now = base::Time::Now().LocalMidnight();
 
   auto ad_history_details = client_->GetAdsShownHistory();
+
+  std::unique_ptr<AdHistoryFilter> ad_history_filter;
+  switch (AdsHistoryFilterType) {
```

Comment:
Can you please raise an issue to refactor to use factories

Reply:
Will do - yep :+1: